main/api/fields.py

```diff
@@ -0,0 +1,46 @@
+# coding: utf-8
+
+import urllib
+
+from flask.ext.restful import fields
+
+
+class BlobKey(fields.Raw):
+  def format(self, value):
+    return urllib.quote(str(value))
+
+
+class Blob(fields.Raw):
+  def format(self, value):
+    return repr(value)
+
+
+class DateTimeField(fields.Raw):
+  def format(self, value):
+    return value.isoformat()
+
+
+class GeoPtField(fields.Raw):
+  def format(self, value):
+    return '%s,%s' % (value.lat, value.lon)
+
+
+class IdField(fields.Raw):
+  def output(self, key, obj):
+    try:
+      value = getattr(obj, 'key', None).id()
+      return super(IdField, self).output(key, {'id': value})
+    except AttributeError:
+      return None
+
+
+class IntegerField(fields.Raw):
+  def format(self, value):
+    if value > 9007199254740992 or value < -9007199254740992:
+      return repr(value)
+    return value
+
+
+class KeyField(fields.Raw):
+  def format(self, value):
+    return value.urlsafe()
```

Comment:
Good question. For such cases Flask-RESTful has ```fields.Nested```. For example:

``` python
class A(ndb.Model):
  b = ndb.FloatProperty(default=12.12)
  bb = ndb.IntegerProperty(default=12)
  bbb = ndb.StringProperty(default='bbb')

class Config(model.Base, model.ConfigAuth):
  ...
  test_a = ndb.StructuredProperty(A)
  def _pre_put_hook(self):
    self.test_a = A()

A_FIELDS = {
  'b': fields.Float,
  'bb': fields.Integer,
  'bbb': fields.String,
}


CONFIG_FIELDS = {
  ...
  'test_a': fields.Nested(A_FIELDS)
}
```
...
![screen shot 2015-01-01 at 21 56 59](https://cloud.githubusercontent.com/assets/47926/5593034/391af01a-9201-11e4-99c6-ea20dbcceb43.png)